plugins/modules/system/iptables_state.py

```diff
@@ -0,0 +1,638 @@
+    feature to avoid this, by telling the host to restore previous rules if a
+    cookie is still there after a given delay, and all this time telling the
+    controller to try to remove this cookie on the host through a new
+    connection.
+notes:
+  - The rollback feature is not a module option and depends on task's
+    attributes. To enable it, the module must be played asynchronously, i.e.
+    by setting task attributes I(poll) to I(0), and I(async) to a value less
+    or equal to C(ANSIBLE_TIMEOUT). If I(async) is greater, the rollback will
+    still happen if it shall happen, but you will experience a connection
+    timeout instead of more relevant info returned by the module after its
+    failure.
+options:
+  counters:
+    description:
+      - Save or restore the values of all packet and byte counters.
+      - When I(True), the module is not idempotent.
+    type: bool
+    default: false
+  ip_version:
+    description:
+      - Which version of the IP protocol this module should apply to.
+    type: str
+    choices: [ ipv4, ipv6 ]
+    default: ipv4
+  modprobe:
+    description:
+      - Specify the path to the modprobe program internally used by iptables
+        related commands to load kernel modules.
+      - By default, /proc/sys/kernel/modprobe is inspected to determine the
+        executable's path.
+    type: path
+  noflush:
+    description:
+      - For I(state=restored), ignored otherwise.
+      - If I(False), restoring iptables rules from a file flushes (deletes)
+        all previous contents of the respective table(s). If I(True), the
+        previous rules are left untouched (but policies are updated anyway,
+        for all built-in chains).
+    type: bool
+    default: false
+  path:
+    description:
+      - The file the iptables state should be saved to.
+      - The file the iptables state should be restored from.
+      - Required when I(state=saved) or I(state=restored).
+    type: path
+  state:
+    description:
+      - Whether the firewall state should be saved (into a file) or restored
+        (from a file).
+      - When this option is not set, the current iptables state is returned.
+    type: str
+    choices: [ saved, restored ]
+  table:
+    description:
+      - When C(state=restored), restore only the named table even if the input
+        file contains other tables.
+      - When C(state=saved) (or left unset), restrict output to the specified
+        table. If not specified, output includes all active tables.
+    type: str
+    choices: [ filter, nat, mangle, raw, security ]
+  wait:
+    description:
+      - Wait N seconds for the xtables lock to prevent instant failure in case
+        multiple instances of the program are running concurrently.
+    type: int
+requirements: [iptables, ip6tables]
+'''
+
+EXAMPLES = r'''
+# This will only retrieve information
+- name: get current state of the firewall
+  iptables_state:
```

Comment:
Examples for new modules must always use FQCNs (Fully Qualified Collection Names):
```suggestion
  community.general.iptables_state:
```